payments/index.ts

```diff
@@ -0,0 +1,80 @@
+import { TransactionInput } from "./types"
+import Stripe from "stripe"
+import mongoDB, { MongoClient } from "mongodb"
+
+import * as dotenv from "dotenv"
+dotenv.config()
+import express from "express"
+
+const app = express()
+const port = 5000
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
+	apiVersion: "2022-11-15",
+	typescript: true,
+})
+const collections: { paymentintents?: mongoDB.Collection } = {}
+
+// function to connect to MongoDB
+async function connectToDatabase() {
+	const client: mongoDB.MongoClient = new MongoClient(
+		process.env.DB_CONN_STRING
+	)
+	await client.connect()
+
+	const db: mongoDB.Db = client.db(process.env.DB_NAME)
+	const collection: mongoDB.Collection = db.collection(
+		process.env.COLLECTION_NAME
+	)
+	collections.paymentintents = collection
+	console.log(
+		`Successfully connected to database: ${db.databaseName} and collection: ${collections.paymentintents}`
+	)
+}
+
+app.get("/", (req, res) => {
+	res.send("Hello World!")
+})
+
+app.listen(port, () => {
+	console.log(`Example app listening on port ${port}`)
+})
+
+// route to create Payment Intent in Stripe
+app.post("/payment", async (req, res) => {
+	const input: TransactionInput = req.query
```

Comment:
currently, the behaviour of this endpoint only works if request is made as:

localhost:5008/payment?amount=<cents>&currency=<currency>

a few things:
- [ ] url path should be pural ("payments" instead of "payment")
- [ ] data passed in a POST request should be in the request body rather than as a request parameter
- [ ] request should also contain `quantity_tco2e`, `project_id`, `owner_id` and `buyer_id` to track who created this payment intent and for which item to be purchased as well as from whom to purchase